static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js

```diff
@@ -187,14 +225,18 @@ const canShow = (): Promise<boolean> => {
         return Promise.resolve(false);
     }
 
-    if (hasUserAcknowledgedBanner(messageCode)) {
-        return Promise.resolve(false);
-    }
-
     bannerParams = deriveBannerParams(getGeoLocation());
+    const hasSeenEnoughArticles =
+        bannerParams && getVisitCount() >= bannerParams.minArticles;
+
+    if (hasSeenEnoughArticles) {
+        return hasBannerBeenRedeployedSinceClosed().then(hasBeenRedeployed => {
+            if (hasBeenRedeployed) {
+                return canDisplayMembershipEngagementBanner();
```

Comment:
in fact, I'm not sure why the `hasSeenEnoughArticles` isn't just inside `canDisplayMembershipEngagementBanner`.

Alternatively, just do away with `canDisplayMembershipEngagementBanner` since it isn't called anywhere else and all it does is all `shouldShowReaderRevenue()` - you could bring that call inside here

Reply:
Done!